pkg/operator/controllers/dnsmasq/cluster_controller.go

```diff
@@ -79,38 +74,38 @@ func (r *ClusterReconciler) Reconcile(ctx context.Context, request ctrl.Request)
 	return reconcile.Result{}, nil
 }
 
-func reconcileMachineConfigs(ctx context.Context, instance *arov1alpha1.Cluster, dh dynamichelper.Interface, roles ...string) error {
+// SetupWithManager setup our mananger
+func (r *ClusterReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	aroClusterPredicate := predicate.NewPredicateFuncs(func(o client.Object) bool {
+		return o.GetName() == arov1alpha1.SingletonClusterName
+	})
+
+	return ctrl.NewControllerManagedBy(mgr).
+		For(&arov1alpha1.Cluster{}, builder.WithPredicates(aroClusterPredicate)).
+		Named(ClusterControllerName).
```

Comment:
Consider a `Watches` for machine config pool resources (if needed)

Reply:
We don't need this as the `cluster_controller` is only watching the ARO Cluster CR, the `machine_config_controller` is watching the machine configs.